sudo permission needed or script fails

running with sudo works
```
10:16:14 andrew Barefoot_WRR_2019:(master)$ sudo make
[sudo] password for andrew: 
./analysis/stony_creek_analysis.r
here() starts at /home/andrew/temp/Barefoot_WRR_2019
Warning message:
In summary.lm(mod) : essentially perfect fit: summary may be unreliable
There were 26 warnings (use warnings() to see them)
```

running without sudo fails, but gives no indication (to the untrained user) why it fails.
```
10:18:32 andrew Barefoot_WRR_2019:(master)$ make
./analysis/stony_creek_analysis.r
here() starts at /home/andrew/temp/Barefoot_WRR_2019
Warning message:
In summary.lm(mod) : essentially perfect fit: summary may be unreliable
Error in gzfile(file, "wb") : cannot open the connection
Calls: source -> withVisible -> eval -> eval -> save -> gzfile
In addition: There were 27 warnings (use warnings() to see them)
Execution halted
makefile:2: recipe for target 'all' failed
make: *** [all] Error 1
```

Also, when running as sudo, I still don't get any folder called `derived data`?

--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,5 @@
 stony_segment_pts_test.xlsx
 
 .DS_Store
+
+pkgs/*

--- a/README.md
+++ b/README.md
@@ -8,7 +8,9 @@ This repository contains code and data to replicate results from Code and data f
 
 You will have to have a working `R` installation to use these routines. To get `R` -- which is free -- download and install from [the R website](https://www.r-project.org/)
 
-This repository only currently runs on UNIX-like OS machines. Sorry Windows people.
+When you install R, there are a number of choices you can make about where libraries are stored. If you choose the defaults, this software should work out of the box. In the event, however, that something is screwy with your installation, all needed packages and dependencies will be downloaded into a `pkgs` directory, which may take some time. Once this process is complete, recompleting the analysis / adding to the code should be much faster. 
+
+This repository only currently runs on UNIX-like machines (OSX and Linux). Sorry Windows people.
 
 To install, download this repository and put it somewhere in your computer. Then open a terminal with this folder as the working directory and type:
 ```bash
@@ -19,20 +21,26 @@ This command will replicate all the analysis presented in the paper, and open dr
 
 To delete all products of the analysis, simply enter `make clean` in the terminal, and all the draft figures will be deleted.
 
+## Bugs and problems
+
+If you encounter an issue installing or running the analysis, or wish to contribute to this software, [please open an issue](https://github.com/ericbarefoot/Barefoot_WRR_2019/issues) describing your problem with a [reproducible example, or `reprex`](https://reprex.tidyverse.org/). reprexes help us help you with your issue, and we'll do our best to fix the bug. If you wish to suggest a feature or similar, please also submit an issue, and the authors will be happy to discuss the feature.
+
 ## Data Availability
 
 [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.1342865.svg)](https://doi.org/10.5281/zenodo.1342865)
 
 Data is also available as .zip files with the DOI listed above. Each zip file corresponds to one subdirectory of the `raw_data` directory (see file tree below).
 
-You do not need to additionally go fetch these data, as they are contained in this repository too. 
+You do not need to additionally go fetch these data, as they are contained in this repository too.
+
+
 
 ## File Structure
 
 If the installation is correct, you should see the following file structure:
 
 ```
-.
+Barefoot_WRR_2019
 ├── analysis
 │   ├── b_calcs.r
 │   ├── compile_data_hl.r
@@ -66,12 +74,16 @@ If the installation is correct, you should see the following file structure:
 │   │   │   ├── stony_creek_event_data.csv
 │   │   │   ├── stony_creek_field_data.csv
 │   │   │   └── stony_creek_strm_order.csv
+│   │   ├── field_data.zip
 │   │   ├── gis_data
-│   │   │   └── stony_segment_pts.dbf
-│   │   └── hydro_data
-│   │       ├── bolin_2012_2017.csv
-│   │       ├── discharge_data.csv
-│   │       └── drain_dates.csv
+│   │   │   ├── stony_segment_pts.dbf
+│   │   │   └── stony_segment_pts_test.xlsx
+│   │   ├── gis_data.zip
+│   │   ├── hydro_data
+│   │   │   ├── bolin_2012_2017.csv
+│   │   │   ├── discharge_data.csv
+│   │   │   └── drain_dates.csv
+│   │   └── hydro_data.zip
 │   ├── README.txt
 │   ├── stony_creek_io.r
 │   └── stony_creek_process.r
@@ -90,8 +102,11 @@ If the installation is correct, you should see the following file structure:
 │       ├── figure_4.r
 │       ├── figure_5.r
 │       └── figure_6.r
+├── library_install.r
 ├── LICENSE
 ├── makefile
+├── pkgs
 └── README.md
 
+12 directories, 54 files
 ```

--- a/analysis/b_calcs.r
+++ b/analysis/b_calcs.r
@@ -36,10 +36,10 @@ for (i in 1:nrow(just_widths)) {
 	logwid = log10(wid_pt[!wz])
 	logQ = log10(sQm[!wz])
 	if (length(which(wz == F)) > 3) {
-		mod = lm(logwid~logQ)
+		mod = suppressWarnings(lm(logwid~logQ))
 		b[i] = mod$coefficients[2]
 		a[i] = mod$coefficients[1]
-		rsqr[i] = summary(mod)$adj.r.squared
+		rsqr[i] = suppressWarnings(summary(mod)$adj.r.squared)
 	} else {
 		b[i] = NA
 		rsqr[i] = NA

--- a/analysis/compile_data_hl.r
+++ b/analysis/compile_data_hl.r
@@ -31,7 +31,7 @@ source(here::here('analysis', 'functions', 'disch_conv.r'))
 
 source(here::here('analysis', 'percentile_calc.r'))
 
-source(here::here('analysis', 'ks_stats.r'))
+suppressWarnings(source(here::here('analysis', 'ks_stats.r')))
 
 #	and some previous data
 

--- a/analysis/stony_creek_analysis.r
+++ b/analysis/stony_creek_analysis.r
@@ -8,22 +8,16 @@
 #
 # OSX/Linux: "./analysis/stony_creek_analysis.r"
 
-########################################################
-# install and load  necessary libraries
 
-packages = c('here', 'foreign', 'lubridate', 'dplyr', 'MASS')
 
-ipak = function(pkg){
-  new_pkg = pkg[!(pkg %in% installed.packages()[, "Package"])]
-  if (length(new_pkg)) {
-    install.packages(new_pkg, dependencies = TRUE)
-  }
-}
+lib = file.path(getwd(), 'pkgs')
+.libPaths(c(lib, .libPaths()))
 
-ipak(packages)
+# message(.libPaths())
+# stop()
+
 
 library(here)
-
 ########################################################
 
 # Data In/Out

--- a/analysis/stony_creek_analysis.r
+++ b/analysis/stony_creek_analysis.r
@@ -8,8 +8,6 @@
 #
 # OSX/Linux: "./analysis/stony_creek_analysis.r"
 
-
-
 lib = file.path(getwd(), 'pkgs')
 .libPaths(c(lib, .libPaths()))
 

--- a/library_install.r
+++ b/library_install.r
@@ -24,6 +24,11 @@ ipak = function(pkg, lib){
   .libPaths(c(lib, .libPaths()))
 
   if (length(new_pkg)) {
+
+    message(paste('You need a few packages, installing... \nThis may take a few minutes depending on how many are missing, \nso you might want to go get coffee.'))
+
+    Sys.sleep(6)
+
     install.packages(new_pkg, dependencies = TRUE, lib = lib, repo = 'https://cloud.r-project.org/')
   }
 

--- a/library_install.r
+++ b/library_install.r
@@ -1,0 +1,32 @@
+#!/usr/bin/env Rscript
+
+# correcting dependency install issues for linux use where library path isn't specified, and a local path is requested
+# Eric Barefoot
+# Aug 2018
+
+# message(getwd())
+
+########################################################
+# install and load  necessary libraries
+
+packages = c('here', 'foreign', 'lubridate', 'dplyr', 'MASS')
+
+lib = file.path(getwd(), 'pkgs')
+
+ipak = function(pkg, lib){
+
+  new_pkg = pkg[!(pkg %in% installed.packages()[, "Package"])]
+
+  if (!file.exists(lib)) {
+    suppressWarnings(dir.create(lib))
+  }
+
+  .libPaths(c(lib, .libPaths()))
+
+  if (length(new_pkg)) {
+    install.packages(new_pkg, dependencies = TRUE, lib = lib)
+  }
+
+}
+
+ipak(packages, lib)

--- a/library_install.r
+++ b/library_install.r
@@ -24,7 +24,7 @@ ipak = function(pkg, lib){
   .libPaths(c(lib, .libPaths()))
 
   if (length(new_pkg)) {
-    install.packages(new_pkg, dependencies = TRUE, lib = lib)
+    install.packages(new_pkg, dependencies = TRUE, lib = lib, repo = 'https://cloud.r-project.org/')
   }
 
 }

--- a/makefile
+++ b/makefile
@@ -1,9 +1,12 @@
 all:
-	time ./library_install.r
+	./library_install.r
 	./analysis/stony_creek_analysis.r
 
 clean:
-	rm data/derived_data/*
-	rm figures/outputs/*_draft*
+	rm -rf data/derived_data/*
+	rm -rf figures/outputs/*_draft*
 
-.PHONY: all clean
+empty_pkgs:
+	rm -rf pkgs/*
+
+.PHONY: all clean empty_pkgs

--- a/makefile
+++ b/makefile
@@ -1,4 +1,5 @@
 all:
+	time ./library_install.r
 	./analysis/stony_creek_analysis.r
 
 clean:

